Add support for uploading proguard mapping files

tmtrademarked opened the pull request:
Thanks for contributing to _fastlane_! Before you submit your pull request, please make sure to check the following boxes:

- [ X ] Run `bundle exec rspec` from the subdirectory of each tool you modified. Alternatively, run `rake test_all` from the root directory.
- [ X ] Run `bundle exec rubocop -a` to ensure the code style is valid
- [ X ] Read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [ X ] We currently don't accept new actions, please publish a plugin instead, more information in [Plugins.md](https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Plugins.md)

Before submitting a pull request, we appreciate if you create an issue first to discuss the change :+1:

This PR adds functionality to support adding a proguard mapping file per apk. This will allow developers to upload their deobfuscation files at the same time as they upload their APKs, making life better for everyone.

This is an attempt to implement the functionality requested in https://github.com/fastlane/fastlane/issues/7183.

Note that I haven't tested this functionality yet - any comments on the best way to write unit tests for this and to replace my currently installed Supply with my locally built version would be much appreciated!

tmtrademarked committed:
Add support for uploading proguard mapping files

This PR adds functionality to support adding a proguard mapping file per apk. This will allow developers to upload their deobfuscation files at the same time as they upload their APKs, making life better for everyone.

This is an attempt to implement the functionality requested in https://github.com/fastlane/fastlane/issues/7183.

Note that I haven't tested this functionality yet - any comments on the best way to write unit tests for this and to replace my currently installed Supply with my locally built version would be much appreciated!

tmtrademarked committed:
Add support for uploading proguard mapping files

This PR adds functionality to support adding a proguard mapping file per apk. This will allow developers to upload their deobfuscation files at the same time as they upload their APKs, making life better for everyone.

This is an attempt to implement the functionality requested in https://github.com/fastlane/fastlane/issues/7183.

Note that I haven't tested this functionality yet - any comments on the best way to write unit tests for this and to replace my currently installed Supply with my locally built version would be much appreciated!

joreilly commented:
Really looking forward to this being available!

hjanuschka commented:
@tmtrademarked 

Maybe one of the below option's help you to run your modified supply.

# Option 1
get this script into your project folder, 
https://gist.github.com/hjanuschka/3040fc6f371f46acff72a5e67f649b27

This requires you to push changes first to github and then call `try_pr.sh` again, 
but its an easy way to every one trien out a PR.

```bash
cd /path/to/your/project/
# this only has to be done once.
curl -o try_pr.sh https://gist.githubusercontent.com/hjanuschka/3040fc6f371f46acff72a5e67f649b27/raw/9600a34188949061a96f77c82505efd99ad6f5e4/try_pr.sh
chmod a+x  try_pr.sh


# then call it with (which in this case should show the help screen of your supply):
./try_pr.sh tmtrademarked proguard_files supply --help
```


# Option 2
Totally local.:

```bash
cd /path/to/your/fastlane-clone/
# ensure your branch is checked out
export FASTLANE_LOCAL_DEV=1
bundle update # only once
bundle install #only once
bundle exec supply --help
```


> just to be sure, unsupported, unofficial, use at your own risk! - and review the shell script before you run it.

hjanuschka commented on supply/lib/supply/options.rb:
```diff
@@ -127,7 +127,30 @@ def self.available_options
+                                     optional: true,
+                                     type: Array,
+                                     description: "An array of paths to mapping files to upload",
+                                     short_option: "-mp",
```
this one raises an exception:

```
fastlane_core/lib/fastlane_core/ui/interface.rb:145:in `user_error!': short_option must be a String of length 1 (FastlaneCore::Interface::FastlaneError)
```

you should use something different than, `-mp`

tmtrademarked committed:
Fixing a few errors
* The short options weren’t valid, so assigning new shortcuts.
* Removed the invalid default value for mapping.
* Missing some syntax in the upload - uploading a proguard file requires the APK version code.
* Use the correct media type when uploading the mapping file. (Play API won't accept text/plain, and requires application/octet-stream)

tmtrademarked committed:
Fixing a few errors
* The short options weren’t valid, so assigning new shortcuts.
* Removed the invalid default value for mapping.
* Missing some syntax in the upload - uploading a proguard file requires the APK version code.
* Use the correct media type when uploading the mapping file. (Play API won't accept text/plain, and requires application/octet-stream)

tmtrademarked commented:
Thanks, @hjanuschka - that worked like a charm! In order to minimize noise, I ended up using your "Option 2" approach. Having tested this with my build, I can now verify that the latest PR uploads mapping files to the play console correctly.

PTAL and let me know what you think of the approach.

hjanuschka commented:
@tmtrademarked thx for feedback, glad you got it rolling!
lets wait for the core team, to review it.

hotchpotch committed:
Fix rbenv install message (#7173)

streeter committed:
Add missing backtick (#7170)

danielbowden committed:
Ensure correct line containing Xcode version is used when other output present (#7171)

mfurtak committed:
Add some simple tests for ensure_xcode_version (#7203)

Commit:
Bump supply version (#7204)

mfurtak committed:
Refactor ensure_xcode_version to do more work in Ruby (#7205)

Supplement the tests to cover the error case from #7171

sainttail committed:
Fix call set_info_plist_value resulting in binary plist file (#7188)

hjanuschka committed:
primaryLanguageCode detection fix (#7193)

hjanuschka committed:
platform bug (#7174)

xai3 committed:
[sigh] Not fulfilled the test case written in the comment when provisioning_profiles is Hash (#7154)

Commit:
Update internal dependencies (#7208)

hjanuschka committed:
truncate changelog, feature-request of: #3956 (#7187)

Commit:
Bump deliver version (#7209)

Palleas committed:
Update message when `sonar-runner` isn't installed (#7214)

Commit:
Bump sign version (#7213)

hjanuschka committed:
[spaceship] non-ASCII app name in produce 1/2 (#7210)

* produce non-ascii [spaceship]

* add spec

* feedback

* feedback

hjanuschka committed:
[produce] non-ASCII app name in produce 2/2, fixes #7206 (#7211)

* produce non-ascii [produce]

* feedback

Commit:
Bump spaceship version (#7224)

hjanuschka committed:
[fastlane_core] sensitive flag on config items  (#7169)

* Added the ability to suppress echo when entering sensitive parameters such as passwords.

Commit:
Bump fastlane version (#7228)

andyvanosdale committed:
Allow rejecting app versions (#7152)

Implements https://github.com/fastlane/fastlane/issues/7149

New method reject! on AppVersion. The version must be in a rejectable state,
as notified in the canRejectVersion field.

Commit:
Revert "[spaceship] non-ASCII app name in produce 1/2" (#7232)

Commit:
Bump spaceship version for hotfix (#7233)

chreno committed:
Added the missed fix to fastlane_core fastlane_folder (#7220)

Commit:
Revert "[produce] non-ASCII app name in produce 2/2" (#7241)

Commit:
Update internal dependecies (#7227)

mfurtak committed:
Babosa fix (#7243)

* [spaceship] non-ASCII app name in produce 1/2 (#7210)
* Introduce a BabosaFix class to work-around Unicode problems

mfurtak committed:
Version Bump (#7245)

mfurtak committed:
Handle non-ascii names in produce (#7246)

* [produce] non-ASCII app name in produce 2/2, fixes #7206 (#7211)
* Remove latinization from produce Group, letting spaceship handle it

mfurtak committed:
Remove babosa dependency from fastlane_core (#7248)

And remove the requires that use it

chreno committed:
Moved fastlane_folder to fastlane_core, fixes #5696 (#5825)

* Moved fastlane_folder to fastlane_core, fixed ./.fastlane folder error in deliver detect_values, added some unit tests for find_folders

* Fixed format issues

* Moved available_lanes from fastlane_folder to cli_tools_distributor

* Fixed fastlane_folder module

* Fixed new usages of FastlaneFolder

mfurtak committed:
Version bump (#7250)

KrauseFx committed:
Move GitHub templates into .github sub-directory (#7244)

KrauseFx committed:
Add information about Appfile location (#7252)

mfurtak committed:
Fix the tests allow for interactive? to be for UI not Helper (#7253)

mfurtak committed:
Update internal dependencies (#7247)

Commit:
Bump produce version (#7256)

hjanuschka committed:
Set sensitive flag on action options (passwords, api-keys, api-secrets, confidential urls) (#7235)

hjanuschka committed:
Make keychain_password a sensitive option (#7236)

hjanuschka committed:
Make keychain_password a sensitive option (#7237)

hjanuschka committed:
Make p12_password a sensitive option (#7238)

hjanuschka committed:
Make slack_url a sensitive option (#7239)

Commit:
Correct style of fastlane in readme (#7259)

KrauseFx committed:
[spaceship] Fix alignment of spaceship version number (#7249)

Via https://twitter.com/mokagio/status/803724200611450880

KrauseFx committed:
Remove the "I" in spaceship readme (#7264)

KrauseFx committed:
Update information about https://developerservices2.apple.com API (#7265)

This API doesn't return the certificates and devices any more

milch committed:
Fix carthage code sample (#7269)

tmm1 committed:
fix broken pilot upload (#7270)

tsafrir committed:
Fix removing uploaded screenshot by using sort_order (#7273)

mfurtak committed:
Add VISION.md to document the fastlane philosophy (#7263)

hjanuschka committed:
[fastlane_core] FastlaneCore::Simulator reset by version 1/2 (#7196)

* reset by os version

* change unless to until

* address feedback

hjanuschka committed:
[fastlane] support aliases in actions (#7217)

* support aliases

* add spec

* add spec

* get alias callback

* move to runner

* rename

* more specs

* Forgotten comma

* lazy load aliases, add reset method

sebastianludwig committed:
Added comment to keep track when the babosa fix can be removed again. (#7277)

KrauseFx committed:
Update fastlane download link in README (#7268)

* Update fastlane download link in README

Let's wait a few hours before merging this so the DNS changes are available world-wide

* Update download link

KrauseFx committed:
Show GitHub issue search error only in verbose mode (#7279)

Fixes confusion, e.g. https://github.com/fastlane/fastlane/issues/7274

KrauseFx committed:
Improve boolean prompt action (#7284)

Now the user doesn’t have to click “Enter” to confirm boolean values any more, using our FastlaneCore UI instead

KrauseFx committed:
Update actions URL to new docs page (#7297)

mgrebenets committed:
Fix error in resign.sh bash syntax (#7302)

Commit:
Bump sigh version (#7305)

mpirri committed:
Create CORE_CONTRIBUTOR.md (#7304)

KrauseFx committed:
Update Gemfile setup to use local copy by default (#7296)

* Update Gemfile setup to use local copy by default

This makes it easier for new fastlane contributors to start changing the code base of fastlane

* Update tests to work with new bundler setup

KrauseFx committed:
Automate fastlane version bumps (#7287)

* Automate fastlane version bumps

- Show changelog
- Ask for type of version bump
- Increment the version number
- Commit the changes
- Push to GitHub
- Submit PR
- Add git label

* Post message to slack after submitting the PR

* Fix code style

* Revert to master branch after submitting the PR

* Change order of checking for GitHub status

* Improve PR Body text

KrauseFx committed:
Make the lane context dump on crashes more pretty (#7285)

* Make the lane context dump on crashes more pretty

* Fix code style

KrauseFx committed:
Add ToolingAndDebugging.md to help with debugging fastlane (#7278)

* Initial commit for ToolingAndDebugging.md

* Fix file URL

* Use HTML tags for screenshots

* Update layout

* Update layout

* Update layout

* Update layout

* Update layout

KrauseFx committed:
Update spaceship to work with the new iTunes Connect (#7311)

KrauseFx committed:
[spaceship] Version bump to 0.38.5 (#7312)

KrauseFx committed:
Update spaceship dependency (#7314)

bassrock committed:
[Pilot] Adding in platform support for pilot (#4449)

* Adding in platform support for pilot

KrauseFx committed:
Move all fastlane configs to ~/.fastlane (#7298)

* Add new FastlaneCore.fastlane_user_dir method

* Migrate .did_show_opt_info into .fastlane sub-directory

KrauseFx committed:
Fix bundler env after recent change with Gemfiles (#7313)

Before that, the following error was shown

```
[11:52:27]: Exit status of command 'bundle install' was 1 instead of 0.
/Users/fkrause/Developer/fastlane/spaceship/Gemfile:2:in `read':  (Bundler::Dsl::DSLError)
[!] There was an error parsing `Gemfile`: No such file or directory @ rb_sysopen - Gemfile. Bundler cannot continue.

 #  from /Users/fkrause/Developer/fastlane/spaceship/Gemfile:2
 #  -------------------------------------------
 #  Dir.chdir("..") do
 >    eval File.read("Gemfile")
 #  end
 #  -------------------------------------------
	from /Users/fkrause/Developer/fastlane/spaceship/Gemfile:2:in `block in eval_gemfile'
	from /Users/fkrause/Developer/fastlane/spaceship/Gemfile:1:in `chdir'
	from /Users/fkrause/Developer/fastlane/spaceship/Gemfile:1:in `eval_gemfile'
	from /Users/fkrause/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bundler-1.13.3/lib/bundler/dsl.rb:41:in `instance_eval'
	from /Users/fkrause/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bundler-1.13.3/lib/bundler/dsl.rb:41:in `eval_gemfile'
	from /Users/fkrause/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bundler-1.13.3/lib/bundler/dsl.rb:11:in `evaluate'
	from /Users/fkrause/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bundler-1.13.3/lib/bundler/definition.rb:33:in `build'
	from /Users/fkrause/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bundler-1.13.3/lib/bundler.rb:127:in `definition'
	from /Users/fkrause/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bundler-1.13.3/lib/bundler.rb:93:in `setup'
	from /Users/fkrause/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bundler-1.13.3/lib/bundler/setup.rb:20:in `<top (required)>'
	from /Users/fkrause/.rbenv/versions/2.2.1/lib/ruby/site_ruby/2.2.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /Users/fkrause/.rbenv/versions/2.2.1/lib/ruby/site_ruby/2.2.0/rubygems/core_ext/kernel_require.rb:55:in `require'
```

This way we'll ensure the local Gemfile isn't used for our docs

mfurtak committed:
Update links in plugins readme template for docs.fastlane.tools (#7300)

* Update links in plugin readme template for docs.fastlane.tools

* Put last fastlane.tools link back

* And fix the link text

KrauseFx committed:
[fastlane_core] Version bump to 0.58.0 (#7318)

KrauseFx committed:
Verify debug code is not there, also in spec sub-folder (#7316)

KrauseFx committed:
Remove `now:` from tests (#7317)

KrauseFx committed:
Update fastlane_core dependencies (#7320)

KrauseFx committed:
[match] Version bump to 0.11.1 (#7288)

KrauseFx committed:
[deliver] Version bump to 1.16.1 (#7286)

KrauseFx committed:
Delete GitHubLabels.md (#7321)

This is now part of https://github.com/fastlane/fastlane/pull/7280

KrauseFx committed:
Fix spelling of fastlane (#7319)

KrauseFx committed:
[pem] Version bump to 1.4.1 (#7289)

hjanuschka committed:
[fastlane] FastlaneCore::Simulator reset by version, fixes #7192 (#7197)

* reset by version

* modify description

* rename and alias action

* fix example

* rubocop

hjanuschka committed:
[fastlane_core] support Commandline apps (#7090)

* classify project

* feedback

hjanuschka committed:
[docs] move AvailablePlugins (#7098)

* move AvailablePlugins

* Update AvailablePlugins.md

hjanuschka committed:
add support for mac apps (#7180)

hjanuschka committed:
[produce] 2/2 produce create mac app, fixes #7166 (#7181)

* support macapps

* also support mac on dev_center

* tunes client filter by platform

milch committed:
Add platform parameter to spaceship iTC client (#7328)

KrauseFx committed:
[s3] Remove duplicate :acl access (#7331)

tmm1 committed:
Teach pilot about apps with multiple platforms (#7267)

* teach pilot about multiple platforms

* use the build platform when updating a build testing status

* pilot is supported on tvos too

* style

* build might be nil

* redundant config

* platform should not be required to list builds

* optional platform for distributing latest build (not from upload)

milch committed:
Try more combinations of keychain paths (#7276)

* Try more combinations of keychain paths

* Make the keychain_paths array easier to modify

Commit:
Fix match keychain tests

Commit:
Fix fastlane_core sierra keychain lookup

dcordero committed:
Add Portal module name when using the Developer Portal interface (#7030)

tmtrademarked commented:
Hey, all - just wanted to check in on the status here. Any thoughts about the feature in general, or the implementation in specific?

asfalcone commented:
@tmtrademarked Thanks for the PR. The idea is great and the code is nice and simple. I am hoping to find some time to pull down the code and test it out before approving. 

I'd like to see a `default_value` for the mapping path. 

If there are other folks who can manually test this, that would be super helpful!

ohayon committed:
[fastlane_core] add support for reporting installation method to refresher (#7348)

ohayon committed:
[fastlane_core] Version bump to 0.59.0 (#7367)

ohayon committed:
update fastlane_core version and remove the copy to clipboard of the release url (#7369)

tmtrademarked commented:
Thanks for the comment, @asfalcone ! Re: default value - the reason I didn't add one here was that I thought it would be somewhat complicated. It's pretty easy to look in the default path for app/release for a mapping.txt file, but if that file doesn't exist, it will cause the "file exists" validation to fail. Some developers may not be using proguard, for example.

So it seemed better to ask developers to explicitly opt in to the mapping file upload by specifying the path. Is there a better way to do the validation that could distinguish the default from a user provided option?

hjanuschka committed:
[gym] support Commandline apps 💪💻 2/2 (#7207)

* cmdline gym builder

* add tvos sample

* add tvos sample

* feedback

* feedback

* Fix #7190

* simplfy sample

* remove emoji alias

* improve building

* user_error if .app not found

milch committed:
Add docs to help the community respond to issues/PRs (#7280)

* Add docs to help the community respond to issues/PRs

* Linkage

* Double line -> Single line + bring code in line with #7278

* List Formatting + quick PR checkout

milch committed:
Add Gemfile for testing (#7371)

* Add Gemfile for testing

* Add link to CONTRIBUTING.md

mfurtak committed:
Fix xcodebuild action specs (#7406)

* Fix xcodebuild action specs

As of 2016-12-7, CircleCI has started setting the FL_BUILDLOG_PATH
environment variable, which changes the default output location for
build logs.

Many specs were assuming that the generated commands
would include the default log path, rather than using the buildlog_path
helper function that takes into account the environment variable.

This fixes those specs to do the right thing.

* Adjust the FL_BUILDLOG_PATH on CircleCI to be the original expected location

milch committed:
Change CONTRIBUTING.md to act as a 'table of contents' for the new contributor docs (#7393)

* Change contributing to act as a 'table of contents' for the new contributor docs

* Surround all fastlanes with _

* So long and thanks for all the fish

milch committed:
Add docs to better explain what goes into a PR (#7392)

* Add docs to better explain what goes into a PR

* Fix doubly empty lines

* Address review

* Fix xcodebuild action specs (#7406)

* Fix xcodebuild action specs

As of 2016-12-7, CircleCI has started setting the FL_BUILDLOG_PATH
environment variable, which changes the default output location for
build logs.

Many specs were assuming that the generated commands
would include the default log path, rather than using the buildlog_path
helper function that takes into account the environment variable.

This fixes those specs to do the right thing.

* Adjust the FL_BUILDLOG_PATH on CircleCI to be the original expected location

* Change CONTRIBUTING.md to act as a 'table of contents' for the new contributor docs (#7393)

* Change contributing to act as a 'table of contents' for the new contributor docs

* Surround all fastlanes with _

* So long and thanks for all the fish

* Add docs to better explain what goes into a PR

* Fix doubly empty lines

* Address review

milch committed:
Fix fastlane_require edgecases, fixes #6951 (#7413)

* Fix built in gems not found

* Remove wrong info about bundle location

* Fix spec name != require name

👀 at you, posix/spawn

* Update fastlane_require.rb

hjanuschka committed:
Identify imessage screenshots, fixes #7366 (#7399)

* identify imessage screenshots

* feedback - change to is_imessage

hjanuschka committed:
Download iMessage screens in right folder, fixes #7366 (#7400)

* download imessage screens in right folder

* feedback - change to is_imessage

hjanuschka committed:
[spaceship] upgrade fastimage, fixes #7370 (#7401)

* upgrade fastimage

* feedback

hjanuschka committed:
[deliver] upgrade fastimage, fixes #7370 (#7402)

* upgrade fastimage

* feedback

hjanuschka committed:
[frameit] upgrade fastimage, fixes #7370 (#7403)

* upgrade fastimage

* feedback

hjanuschka committed:
[snapshot] upgrade fastimage, fixes #7370 (#7404)

* upgrade fastimage

* feedback

sebastianludwig committed:
Remove Shenzhen dependency (#6991)

* Moved Shenzhen plugin functionality into fastlane actions and removed Shenzhen dependency

* Added explicit faraday dependency to fastlane.gemspec

* Added explicit faraday dependency to fastlane.gemspec

* Update spaceship dependency (#7314)

KrauseFx committed:
[deliver] Remove accessing watchbuild config (#7437)

hjanuschka committed:
Env Class, fixes #7326 (#7421)

* [fastlane_core] Split\ of\ \#7337

* add files

* fix require order

milch committed:
Update docs to point to the testing Gemfile (#7372)

hjanuschka committed:
[fastlane_core] improve doc (#7364)

* improve doc

* improve again

* remove tap notice

* add brew update

* feedback

milch committed:
Fix rubocop warning (#7441)

It was likely intended that that case should fall through to the next block, but that isn’t something that ruby does

milch committed:
Fix multiline trailing if warning (#7442)

hjanuschka committed:
[deliver] no screenshot information - reword (#7390)

* reword error msg

* Remove duplicate space

milch committed:
Fix rubocop trailing multiline if warning (#7443)

milch committed:
👮🏽 (#7445)

GoodMirek committed:
Advice how to solve dependency errors (#7451)

Advice how to solve dependency errors, i.e. `bundle update`

KrauseFx committed:
Add explicit names to all tools (#7450)

SiarheiFedartsou committed:
wrong link to tooling and debugging tips fixed (#7416)

* wrong link to tooling and debugging tips fixed

* Trigger CI

milch committed:
[spaceship] Version bump to 0.39.0 (#7456)

milch committed:
[fastlane_core] Version bump to 0.60.0 (#7454)

milch committed:
Update internal dependencies (#7463)

milch committed:
Disable rake < 12 in gem specs (#7464)

* Update internal dependencies

* Use rake versions < 12

milch committed:
[cert] Version bump to 1.4.5 (#7462)

milch committed:
[pilot] Version bump to 1.13.0 (#7461)

milch committed:
[gym] Version bump to 1.13.0 (#7460)

milch committed:
[snapshot] Version bump to 1.17.0 (#7459)

milch committed:
[deliver] Version bump to 1.16.2 (#7458)

milch committed:
[produce] Version bump to 1.4.0 (#7457)

milch committed:
Update fastlane dependencies (#7465)

milch committed:
[fastlane] Version bump to 1.112.0 (#7455)

KrauseFx committed:
[pilot] Fix setting of build information (#7473)

Fixes https://github.com/fastlane/fastlane/issues/7439

milch committed:
[pilot] Version bump to 1.13.1 (#7474)

milch committed:
Bump rubocop dependency (#7444)

RishabhTayal committed:
added option for showing different message when using dsym only option (#7493)

RishabhTayal committed:
Update hockey.rb (#7498)

dcordero committed:
Include appletvos platform when trying to find the app (#7488)

milch committed:
Add message explaining fastlane 2.0 usage (#7507)

KrauseFx committed:
Update gemspecs for fastlane 2.0 (#7452)

* Update gemspecs for fastlane 2.0

* Show post install message

* Colorize post install output

* Remove circular credentials manager deps

* Add rake to Gemfile

* Re-add development dependencies

KrauseFx committed:
Version bump for all tools for pre-2.0 (#7509)

KrauseFx committed:
Include fastlane version in core (#7511)

KrauseFx committed:
fastlane_core version bump (#7512)

KrauseFx committed:
Fix fastlane version (#7514)

* Include fastlane version in core

* Hotfix for receiving the fastlane version

This also includes patch version bumps

KrauseFx committed:
Fix pem code style (#7515)

KrauseFx committed:
Migrate fastlane to be a mono gem (#7436)

* Remove license files from sub-directories

* Remove version files from sub-directories

* Remove gemspecs

* Remove Gemfiles

* Migrate executables

* Update root rake files

* Add new bin dir

* Version bump

* New gemspec

* Add proxy for all tools

* Show warning in bin proxy

* Add explicit names to all tools

* Restore fastlane executable

* Fix fastlane bin stub

* Change all versions to use the fastlane version

* Add spaceship playground support

* Fix code style

* Update rspec dependency

* Move out ::DESCRIPTION into main module definition

* Remove watchbuild from main repo

* Remove danger device grid plugin

* Add plugins to fastlane

* Move description back to fastlane version.rb

Otherwise all of fastlane will have to be pulled in just to include version information

* Command generator for spaceship / fix spaceship executable for monogem (#7479)

* implement commands_generator for spaceship

* Remove watchbuild

* Fix for spaceauth executable

* Pass parrameter list to exec

Otherwise `spaceauth -u abc@a.de` thinks `-u abc@a.de` is a single param

* Add comment to explain  exec invocation

* Add fastlane-credentials bin

* Update install instructions to use `fastlane` instead of other gems (#7496)

* Update install instructions to use `fastlane` instead of other gems

* Update snapshot, sigh, match and gym instructions for mono gem

* Update supply, spaceship, screengrab, scan, produce and pilot usage instructions

* Update pem and gym instructions

* Update readmes

* Refactor mono gem tests to be mono also (#7478)

Refactor mono gem tests to be mono also

* Improve wording of binary error message

* Add support for emoji binaries

* Update update_fastlane action to only update the fastlane gem

* Add new `exclude` option to `ensure_no_debug_code` action

* Update the fastlane deployment pipline to work with the mono gem

* Fix `fastlane bump` for mono gem

* Add Pluginfile

* Update persistence file path for failed rspec tests

* Update auto-rubocop-config copy script for fastlane plugins

* Improve fastlane dependencies

* Add comment to plugin’s default gemspec to avoid circular dependencies

* Fix shell escaping for parameters with space

* Fix snapshot report generation

This was still using the old API to fetch an asset file

* Add missing dotenv dependency

* Enable fastlane releases

* Simplify json dependency

* Remove binary proxies for initial fastlane 2.0 release

* Update tests for fastlane version with no binaries for sub-tools

* Fix code style

* Add rb-readline as development dependency to fix pry issues

* Use rake to run tests

* Version bump

* [fastlane] Version bump

* Fix action collector specs

KrauseFx committed:
Show git command on verbose mode when deploying fastlane (#7516)

milch committed:
Add fastlane/version imports (#7520)

milch committed:
[fastlane] Version Bump (#7521)

KrauseFx committed:
[gym] Fix not archiving builds at all (#7526)

Fixes https://github.com/fastlane/fastlane/issues/7485
Fixes https://github.com/fastlane/fastlane/issues/7494

KrauseFx committed:
Version bump (#7532)

In particular to fix gym https://github.com/fastlane/fastlane/pull/7526

KrauseFx committed:
Losen up rubyzip dependency (#7525)

- Fixes #7517
- Fixes #7534

hjanuschka committed:
fix crash (#7543)

KrauseFx committed:
Add missing require statement (#7544)

* Add missing require statement

Fixes https://github.com/fastlane/fastlane/issues/7501#issuecomment-267599347

I don’t understand why this is necessary, and I can’t reproduce it locally. Maybe something else is wrong

* Move require to beginning of file

KrauseFx committed:
Version bump to 2.0.5 (#7545)

* Version bump to 2.0.5

* Fix Fastfile version bump

KrauseFx committed:
Add note about regressions (#7531)

KrauseFx committed:
Add fastlane `rake release` task to push new fastlane releases (#7528)

KrauseFx committed:
Delete Rakefiles from sub-directories (#7529)

stevenyap committed:
Allow android platform for testfairy action (#7500)

KrauseFx committed:
Disable auto-install and show message when user uses bundler and missing gem (#7527)

* Disable auto-install and show message when user uses bundler and misses a gem

* Revert "Disable auto-install and show message when user uses bundler and misses a gem"

This reverts commit f4235c3fd675dc4e55bd261a35a36c3015aa3bcd.

* Disable auto-install and show message when user uses bundler and misses a gem

* Fix tests after recent change

KrauseFx committed:
Change output directory for tests to show them in circle (#7530)

hjanuschka committed:
[deliver] support app_icons via metadata (#7398)

* support app_icons via metadata

* feedback

GoodMirek committed:
Added action sh, fixing #182 (#7453)

* Added action sh, fixing #182

* Improve wording

* Smaller clean-ups

* Fix tests

hjanuschka committed:
[fastlane] more tests for aliases (#7546)

* more tests

* Update action_spec.rb

hjanuschka committed:
user_error if plist error (#7547)

hjanuschka committed:
[gym] Remove promotion for use_legacy_build_api (#7379)

* remove legace notes

* feedback

* fix readme

hjanuschka committed:
fix platform (#7360)

RishabhTayal committed:
Updated docs description for pilot/testflight action (#7550)

* Update hockey.rb

* clear explanation of `distribute_external` not working while using skip waiting for processing option

nikolaykasyanov committed:
Add "--fail-on-errors" support to Danger action (#7487)

* Add "--fail-on-errors" support to Danger action

* Remove unnecessary line break from parameter description

KrauseFx committed:
Fix not showing the lane list and selection when not passing any arguments (#7549)

Introduced with https://github.com/fastlane/fastlane/pull/7436/commits/7e40156468bb18fe7f51666bb14a429a522c845e with wrongly nested if/else blocks

cc @milch

mdarnall committed:
Update README.md (#7385)

* Update README.md

Added some additional examples using proxies and iTunes Transporter

* Update README.md

Switch ordering of text to _after_ the java doc links

* Trigger Circle

KrauseFx committed:
Avoid loading and showing the Snapfile loading twice (#7497)

lacostej committed:
Not all specs are enabled since the move to monogem (#7559)

* Re-enable all specs (after move to monogem)

* Fix action tests when third party gem isn't installed

KrauseFx committed:
Version bump to 2.1.0 (#7555)

KrauseFx committed:
Remove xcode-install dependendy from fastlane (#7554)

ffittschen committed:
Fix typo which causes a crash (#7569)

milch committed:
Version bump to 2.1.1 (#7572)

* Version bump to 2.1.1

* Add xcode-install to Gemfile

giginet committed:
Apply new calling method to match README (#7583)

tmtrademarked commented:
Quick ping on this one - any further refinements/alterations/etc people would like to see?

hjanuschka committed:
[supply] Split\ of\ \#7337 (#7427)

hjanuschka committed:
[snapshot] ENV class (#7426)

hjanuschka committed:
[fastlane] ENV class (#7422)

* [fastlane] Split\ of\ \#7337

* revert

hjanuschka committed:
[gym] ENV class (#7423)

hjanuschka committed:
[match] ENV class (#7424)

* [match] Split\ of\ \#7337

* revert pw

hjanuschka committed:
[scan] ENV class (#7425)

* [scan] Split\ of\ \#7337

* feedback

philipp-heyse committed:
Fixed parallel upload locked due to temp file in ipa_file_analyser (#7359)

* Fixed parallel upload locked due to temp file in ipa_file_analyser

* replacing hashed tmp file with a tmp dir solution

* replacing hashed tmp file with a tmp dir solution

* fixing merge

* Correct rubocop warnings

* Improve code style

KrauseFx committed:
[spaceship] Show update message on spaceship failure (#7325)

ohayon committed:
Revert "[spaceship] Show update message on spaceship failure" (#7589)

milch committed:
Refactor IpaFileAnalyser (#7587)

hjanuschka committed:
[gym] temp. disable `produces_archives?` to fix user issues. (#7591)

* [gym] temp. disable `produces_archives?` to fix user issues.

* remove FIXME notes

* address feedback

* revert marker too

KrauseFx committed:
Version bump to 2.1.2 (#7596)

hjanuschka committed:
[spaceship] adopt to new API protocol for 2FA (#7575)

* [spaceship] adopt to new API protocol for 2FA

* [spaceship] 2fa api change, check for array and later on first element

ohayon committed:
fixnum is never going to be a string (#7600)

otaran committed:
[actions] hockey: add option to bypass CDN (#7598)

`https://upload.hockeyapp.net` is a CDN, which reports errors occasionally (see #6101).
Using `https://rink.hockey.net` may be slower outside of US, therefore it is turned off by default.

mpirri committed:
Version bump to 2.1.3 (#7603)

hjanuschka committed:
[snapshot] add additional ipad classifiers. (#7570)

hjanuschka committed:
[fastlane] anon. the sensitive options (#7335)

* anon. the sensitive options

* feedback

* feedback

ohayon committed:
iCell mailgun (#7605)

* support mailgun replyto

* fix MAILGUN_REPLY_TO misspell

* not send reply_to if it is nil

* change hash set value method

* fix syntax error

* checkout info.plist

* fix reply to key error

* rubocop

ohayon committed:
[spaceship] fix path to referenced fastfile (#7604)

* fix path to referenced fastfile

* change all the paths

* remove empty Fastfiles

Commit:
Setting --default-path for rspec and made all testing directions accurate (#7607)

milch committed:
Update Gemfile for monogem (#7612)

Fixes #7610

hjanuschka committed:
Revert library support of gym, fixing archive errors, fixes #7630 (#7631)

milch committed:
Version bump to 2.2.0 (#7632)

marcelofabri committed:
Preventing actions called from another action to show in summary (#6322)

* Preventing actions called from another actino to show in summary

Check #4546 for more details

* Add tests 🚀

lacostej committed:
Update CORE_CONTRIBUTOR.md (#7634)

lacostej committed:
Use Dir.mktmpdir internal cleanup mechanism to remove tmp folder (#7635)

lacostej committed:
Jenkins doc improvements (#7608)

* Better explain why not deploying at each commit

* Jenkins / git / changelog integration

hjanuschka committed:
[fastlane] Remove imagemagick check, fixes #6904 (#6905)

* also support gm - as badge uses minimagick

* remove spec - as check is inside other gem

* remove imageamgick checks

hjanuschka committed:
[deliver] support metdata update of live_version (#7172)

* support edit of live version

* allow price tier update
code consistent

* typo

* typo

* feedback

hjanuschka committed:
[README] spaceauth - adopt to monogem changes (#7615)

* [README] spaceauth - adopt to monogem changes

* add spaceship as a tool

* Update README.md

lacostej committed:
[docs] More diagnostics & docs on the DEBUG env during testing (#7560)

* Allow to force the output during testing if DEBUG is set

* Also explain where the stdout output is when DEBUG isn't on

* Consistency in documentation wording WRT DEBUG environment variable

* Remove duplicated stdout redirection

hjanuschka committed:
[snapshot] Add xcargs option to pass additional parameters to xcodebuild, fixes #7255 (#7261)

Requested in: #7255

KrauseFx committed:
[docs] Improve docs around how to test code (#7553)

* Improve docs around how to test code

* Fix spelling mistak

KrauseFx committed:
Update Jenkins integration link in fastlane sub-dir (#7567)

KrauseFx committed:
[docs] Update Jenkins integration link to link to new docs page (#7566)

milch committed:
Updated test path for post-monogem times (#7638)

Commit:
Updates credential manager docs for execution with monogem (#7640)

jdee committed:
Allow commit_version_bump to find the Settings.bundle in the project (#6997)

schung7 committed:
[action] Removed redundant check in reset_git_repo, fixes #7650 (#7649)

milch committed:
Fix broken docs links in ToolsAndDebugging.md (#7637)

milch committed:
Version bump to 2.3.0 (#7654)

joeboyscout04 committed:
Added option for configuring the TEST_TARGET_NAME from snapshot (#7620)

GoodMirek committed:
Description of 'sh' working directory (#7564)

* Description of 'sh' working directory

As discussed in fastlane/fastlane#6982

* Added reference to Advanced.md 'sh' action description

* Changes based on Felix's feedback

* Align style and terminology

KrauseFx committed:
Add note about version in cocoapods (#7642)

We've already received 3 PRs adding this parameter, this should avoid it in the future.

lacostej committed:
junit_generator: prefix index with zeroes to help alpha-numeric sorting (#7657)

KrauseFx committed:
sh action now supports log and error_callback parameters and hide sensitive parameters (#7665)

* sh action now supports log and error_callback parameters

Until now it only worked when using Actions.sh(…).

The method this PR modifies is the alias for Actions.sh

* Hide shell command if `log` is set to false to protect sensitive information

hjanuschka committed:
[fastlane_core] ensure to throw a user error if no xcode is found. (#7671)

* [fastlane_core] ensure to throw a user error if no xcode is found.

* change wording

KrauseFx committed:
Improve wording of require warning (#7666)

hjanuschka committed:
fix json in ruby 2.4 (#7674)

* fix json gemspec

* Add missing space

hjanuschka committed:
Update options.rb (#7684)

KrauseFx committed:
Version bump to 2.3.1 (#7687)

cpunion committed:
Add Spaceship::Portal::App#update_name (#7688)

KrauseFx committed:
Fix spaceship tests shouldn’t include version number (#7693)

laiso committed:
add new param 'disable_notify'  to deploygate-action (#7698)

tomorrowkey committed:
Add Android support to DeployGate action (#6166)

bartoszj committed:
Fixed snapshot reset simulators versions handling. (#7681)

hjanuschka committed:
[snapshot] fix issue with launch arguments during copy of screenshots (#7670)

* [snapsho] fix issue with launch arguments during copy of screenshots

* use dev-name-md5.png as a file name.

GoodMirek committed:
Fix to git command (#7447)

Fix to git command

olegoid committed:
credentials-manager: Fetch port, path, protocol for new internet passwords (#7628)

* [credentials-manager] Fetch port, path, protocol from evn

In case if user want to supply internet passwords added by
credentials-manager with additional metadata(port, path, protocol)
he/she can set environment variables which then will be fetched
in 'add_to_keychain' method passed to security tool.

List of env variables used:
* FASTLANE_PATH - path
* FASTLANE_PORT - port
* FASTLANE_PROTOCOL - protocol

* [credentials_manager] Address code review comments:

* Skip "options" if no evn variables were set
* Add comment describing that "options" used by Xamarin Studio

Vratislav committed:
add sign option to add_git_tag action (#7714)

elliot-nelson committed:
deliver: respect the '--platform' command-line setting (#7648)

asfalcone commented:
Sorry for the delay! I was off for the holidays! 🎄 ⭐️

asfalcone commented on supply/lib/supply/uploader.rb:
```diff
@@ -115,6 +115,15 @@ def upload_binaries
 
+      mapping_paths = [Supply.config[:mapping]] unless (mapping_paths = Supply.config[:mapping_paths])
+      mapping_paths.zip(apk_version_codes).each do |mapping_path, version_code|
+        puts mapping_path
```
Can we remove the `puts` here? 

If you think the output is valuable, provide a nicely formatted string like `UI.message "Uploading #{mapping_path} for #{version_code}."`

tmtrademarked replied:
Whoops - those were just debugging code. Thanks for the reminder! Removed.

tmtrademarked committed:
Removing debugging output

tmtrademarked committed:
Removing debugging output

asfalcone commented:
Looks like we've got some failing tests. Probably just needs to be rebased on master @tmtrademarked

tmtrademarked committed:
Merge branch 'proguard_files' of https://github.com/tmtrademarked/fastlane into proguard_files

tmtrademarked commented:
I thought I did the rebasing correctly, but it looks like I may have messed something up.

tmtrademarked commented:
This looks like it may be tricky to recover from. Let me close this PR and recreate from a fresh fork, which will be simpler. (Any tips on how to do this sort of thing correctly in the future are appreciated, though!)

asfalcone commented:
#7721 replaces this
